src/blockchain/schema.rs

```diff
@@ -24,12 +23,18 @@ use exonum::{
 use log::{error, trace};
 
 use crate::{
-    btc::{BtcAnchoringTransactionBuilder, BuilderError, Sha256d, Transaction},
+    btc::{self, BtcAnchoringTransactionBuilder, BuilderError, Sha256d, Transaction},
     config::Config,
+    proto::BinaryMap,
 };
 
 use super::{data_layout::*, BtcAnchoringState};
 
+/// A set of signatures for a transaction input ordered by the anchoring node identifiers.
+pub type InputSignatures = BinaryMap<u16, btc::InputSignature>;
+/// A set of funding transaction confirmations.
+pub type TransactionConfirmations = BinaryMap<btc::PublicKey, ()>;
```

Comment:
There is a need of `BinarySet` detected 😄 ?

Reply:
I think that such data structures can be useful in many services but in my opinion we have to write some separate crate like `itertools` for them.